File: backend/app/models/sale.py
from sqlalchemy import (
    Column,
    Integer,
    ForeignKey,
    Date,
    Numeric
)
from sqlalchemy.orm import relationship

from app.database.base import Base


class Sale(Base):

    __tablename__ = "sales"

    sale_id = Column(Integer, primary_key=True, index=True)

    sale_date = Column(Date, nullable=False)

    store_id = Column(
        Integer,
        ForeignKey("stores.store_id"),
        nullable=False
    )

    product_id = Column(
        Integer,
        ForeignKey("products.product_id"),
        nullable=False
    )

    quantity_sold = Column(
        Integer,
        nullable=False
    )

    revenue = Column(
        Numeric(10,2),
        nullable=False
    )

    discount_percentage = Column(
        Numeric(5,2),
        default=0
    )

    store = relationship("Store")

    product = relationship("Product")

File: SmartInventory.API/Models/Sale.cs
namespace SmartInventory.API.Models;

public class Sale
{
    public int SaleId { get; set; }

    public int ProductId { get; set; }

    public int StoreId { get; set; }

    public int QuantitySold { get; set; }

    public decimal Revenue { get; set; }

    public DateTime SaleDate { get; set; }

    public Product? Product { get; set; }

    public Store? Store { get; set; }
}